File: android/src/main/java/com/oblador/keychain/decryptionHandler/DecryptionResultHandlerNonInteractive.java
package com.oblador.keychain.decryptionHandler;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.oblador.keychain.cipherStorage.CipherStorage.DecryptionContext;
import com.oblador.keychain.cipherStorage.CipherStorage.DecryptionResult;
import com.oblador.keychain.exceptions.CryptoFailedException;

import java.io.IOException;
import java.security.GeneralSecurityException;
import java.util.concurrent.CompletableFuture;

public class DecryptionResultHandlerNonInteractive implements DecryptionResultHandler {
  @Override
  public CompletableFuture<DecryptionResult> authenticate(@NonNull DecryptionResultJob job) {
    final CompletableFuture<DecryptionResult> result = new CompletableFuture<>();
    try {
      result.complete(job.get());
    } catch (final Throwable fail) {
      result.completeExceptionally(fail);
    }
    return result;
  }
}

File: android/src/main/java/com/oblador/keychain/decryptionHandler/DecryptionResultHandlerNonInteractive.kt
package com.oblador.keychain.decryptionHandler

import androidx.annotation.NonNull
import androidx.annotation.Nullable
import com.oblador.keychain.cipherStorage.CipherStorage.DecryptionContext
import com.oblador.keychain.cipherStorage.CipherStorage.DecryptionResult
import com.oblador.keychain.exceptions.CryptoFailedException

class DecryptionResultHandlerNonInteractive : DecryptionResultHandler {

  // Use 'override' and explicitly declare visibility
  override var result: DecryptionResult? = null
  override var error: Throwable? = null

  override fun askAccessPermissions(@NonNull context: DecryptionContext) {
    val failure = CryptoFailedException("Non-interactive decryption mode.")
    onDecrypt(null, failure)
  }

  override fun onDecrypt(
      @Nullable decryptionResult: DecryptionResult?,
      @Nullable error: Throwable?
  ) {
    this.result = decryptionResult
    this.error = error
  }

  override fun waitResult() {
    // Do nothing, expected synchronized call in one thread
  }
}
